control_panel_api/models.py

```diff
@@ -234,24 +234,37 @@ class Meta:
     def aws_role_name(self):
         return self.app.iam_role_name
 
+    def has_readwrite_access(self):
+        return self.access_level == self.READWRITE
+
 
 class UserS3Bucket(AccessToS3Bucket):
     """
     A user can have access to several S3 buckets.
 
     We have two access levels, "readonly" (default) and "readwrite".
-    The `is_admin` field determine if the user has admin privileges on the
-    S3 bucket
-    """
 
+    The `is_admin` field determines the `access_level` to the S3Bucket. The
+    `access_level` can not be changed directly.
+    """
     user = models.ForeignKey(
         User, related_name='users3buckets', on_delete=models.CASCADE)
     is_admin = models.BooleanField(default=False)
 
     class Meta:
-        # one record per user/s3bucket
         unique_together = ('user', 's3bucket')
         ordering = ('id',)
 
     def aws_role_name(self):
         return self.user.iam_role_name
+
+    def has_readwrite_access(self):
+        return self.is_admin
+
+    def save(self, **kwargs):
+        if self.has_readwrite_access():
```

Comment:
Again, this feels a bit confusing. When saving one of this records, this would sneakingly change the `access_level` even if I didn't ask for it which is a bit surprisingly.

Reply:
Check the ticket please https://trello.com/c/WuFheSAR/447-confirm-users-assigned-s3-bucket-policy-is-read-only-2

Reply:
Yeah I see, I'm a bit confused about that ticket to be honest.